src/components/Table/TableConfirmButtonsRow.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { createPortal } from 'react-dom';
+import { ConfirmButton, CancelButton } from '../InlineEdit';
+import { noop, bindMethods, debounce } from '../../common/helpers';
+
+class TableConfirmButtonsRow extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+
+    bindMethods(this, ['saveRowDimensions', 'handleScroll', 'handleResize']);
+    this.handleScroll = debounce(this.handleScroll, 100);
+    this.handleResize = debounce(this.handleResize, 100);
+  }
+
+  componentDidMount() {
+    this.fetchClientDimensions();
+    window.addEventListener('scroll', this.handleScroll);
+    window.addEventListener('resize', this.handleResize);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.handleScroll);
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  saveRowDimensions(element) {
+    if (element) {
+      this.element = element;
+    }
+    if (this.element) {
+      this.setState({
+        rowDimensions: this.element.getBoundingClientRect()
+      });
+    }
+  }
+
+  handleScroll(event) {
+    this.saveRowDimensions();
+  }
+
+  handleResize(event) {
+    this.fetchClientDimensions();
+    this.saveRowDimensions();
+  }
+
+  fetchClientDimensions() {
+    this.setState({
+      window: {
+        width: document.documentElement.clientWidth,
+        height: document.documentElement.clientHeight
+      }
+    });
+  }
+
+  renderConfirmButtons() {
+    const divStyle = this.state.rowDimensions
+      ? this.props.buttonsPosition(
+          this.state.window,
+          this.element.getBoundingClientRect()
+        )
+      : {};
+
+    const buttonsClass = `inline-edit-buttons ${this.props.buttonsClassName}`;
+    return (
+      <div style={divStyle} className={buttonsClass} key="confirmButtons">
+        <ConfirmButton
+          bsStyle="primary"
+          key="confirm"
+          aria-label={this.props.messages.confirmButtonLabel}
+          onMouseUp={() => this.props.onConfirm()}
+        />
+        <CancelButton
+          bsStyle="default"
+          key="cancel"
+          aria-label={this.props.messages.cancelButtonLabel}
+          onMouseUp={() => this.props.onCancel()}
+        />
+      </div>
+    );
+  }
+
+  render() {
+    const editing = this.props.isEditing();
+    const rowClass = editing ? 'editing' : '';
+
+    const elements = [
+      <tr ref={this.saveRowDimensions} className={rowClass} key="tableRow">
+        {this.props.children}
+      </tr>
+    ];
+
+    if (editing && (this.element || this.props.buttonsMountpoint)) {
+      elements.push(
+        // mount the confirm buttons below the table
+        createPortal(
+          this.renderConfirmButtons(),
+          this.props.buttonsMountpoint ||
+            this.element.closest('table').parentNode
```

Comment:
do any of the downstream products consuming this care about IE11? If so - maybe we just include a [polyfill](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) for element.closest.